File: be/internal/domain/spouse.go
package domain

import "time"

type Spouse struct {
	SpouseID     int        `json:"spouse_id"`
	FatherID     int        `json:"father_id"`
	MotherID     int        `json:"mother_id"`
	MarriageDate *time.Time `json:"marriage_date"`
	DivorceDate  *time.Time `json:"divorce_date"`
	DeletedAt    *time.Time `json:"deleted_at,omitempty"`
}

type SpouseWithMemberInfo struct {
	SpouseID     int        `json:"spouse_id"`
	MemberID     int        `json:"member_id"`
	ArabicName   string     `json:"arabic_name"`
	EnglishName  string     `json:"english_name"`
	Gender       string     `json:"gender"`
	Picture      *string    `json:"picture"`
	MarriageDate *time.Time `json:"marriage_date"`
	DivorceDate  *time.Time `json:"divorce_date"`
}

File: fe/src/types/spouse.ts
export interface Spouse {
  member1_id: number;
  member2_id: number;
  marriage_date: string | null;
  divorce_date: string | null;
}

export interface SpouseCreateInput {
  member1_id: number;
  member2_id: number;
  marriage_date?: string;
  divorce_date?: string;
}



